src/ast.rs

```diff
@@ -39,44 +39,39 @@ pub enum PolymorphismMode {
     Final,
 }
 
-#[derive(Clone, Debug, PartialEq)]
-pub enum DirectAccess {
+#[derive(Copy, Clone, Debug, PartialEq)]
+pub enum DirectAccessType {
     Bit,
     Byte,
     Word,
     DWord,
 }
 
-impl DirectAccess {
+impl DirectAccessType {
     /// Returns true if the current index is in the range for the given type
-    pub fn is_in_range(&self, index: u32, data_type: &DataTypeInformation) -> bool {
-        self.to_bits(index) < data_type.get_size()
+    pub fn is_in_range(&self, index: u64, data_type: &DataTypeInformation) -> bool {
+        (self.get_bit_witdh() * index) < data_type.get_size() as u64
     }
 
     /// Returns the range from 0 for the given data type
-    pub fn get_range(&self, data_type: &DataTypeInformation) -> Range<u32> {
-        0..((data_type.get_size() / self.get_bit_witdh()) - 1)
+    pub fn get_range(&self, data_type: &DataTypeInformation) -> Range<u64> {
+        0..((data_type.get_size() as u64 / self.get_bit_witdh()) - 1)
     }
 
     /// Returns true if the direct access can be used for the given type
     pub fn is_compatible(&self, data_type: &DataTypeInformation) -> bool {
-        data_type.get_size() > self.get_bit_witdh()
+        data_type.get_size() as u64 > self.get_bit_witdh()
     }
 
     /// Returns the size of the bitaccess result
-    pub fn get_bit_witdh(&self) -> u32 {
+    pub fn get_bit_witdh(&self) -> u64 {
         match self {
-            DirectAccess::Bit => 1,
-            DirectAccess::Byte => 8,
-            DirectAccess::Word => 16,
-            DirectAccess::DWord => 32,
+            DirectAccessType::Bit => 1,
```

Comment:
Can we use the corresponding constants from the typesystem?

Reply:
They are not yet commited, should I pull them to this change?